src/lava/magma/runtime/runtime.py

```diff
@@ -40,9 +41,9 @@ def target_fn(*args, **kwargs):
 
 
 class Runtime:
-    """Lava runtime which consumes an executable and run run_condition. Exposes
+    """Lava runtime which consumes an executable and add_ports_for_polling run_condition. Exposes
```

Comment:
I have no idea what this docstring is trying to tell me, especially with this "add_ports_for_polling" word. 

The Lava runtime is an extremely central component of the entire framework. There should be a thoroughly written paragraph or more explaining what the Runtime is, what it does, how it works, how it interacts with other parts of the SW stack, and how it's to be used at a high level.

Imagine you being an outside developer trying to understand Lava, then such docstrings are not very motivating to read on.

Reply:
Also add docstrings to all methods. Even private methods should at least have a one-liner.